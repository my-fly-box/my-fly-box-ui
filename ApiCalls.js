export const fetchFlies = async () => {
  try {
    const response = await fetch(
      "https://my-fly-box-api.herokuapp.com/api/v1/flies"
    );
    const flyData = await response.json();
    return flyData;
  } catch (error) {
    console.error(error);
  }
};

export const addFlyToAPI = async (flyData) => {
  try {
    const response = await fetch('https://my-fly-box-api.herokuapp.com/api/v1/flies', {
      method: 'POST',
      headers: {
        'Content-Type': 'application/json;charset=utf-8'
      },
      body: JSON.stringify(flyData),
    });
    const flyDataResponse = await response.json();
    return flyDataResponse;
  } catch (error) {
    console.error(error);
  }
};

export const addUpdatedFly = async (flyData) => {
  try {
    const response = await fetch(`https://my-fly-box-api.herokuapp.com/api/v1/flies/${flyData.id}`, {
      method: 'PATCH',
      headers: {
        'Content-Type': 'application/json;charset=utf-8'
      },
      body: JSON.stringify(flyData.attributes),
    });
    const flyDataResponse = await response.json();
    return flyDataResponse;
  } catch (error) {
    console.error(error);
  }
}
export const deleteFly = async (flyId) => {
    try {
      const response = await fetch(
        `https://my-fly-box-api.herokuapp.com/api/v1/flies/${flyId}`,
        {
          method: "DELETE",
          headers: {
            "Content-Type": "application/json;charset=utf-8",
          },
        }
      );
      const deletionResponse = await response.json();
      console.log(deletionResponse);
    } catch (error) {
      console.error(error);
    }
  };

  export const addFishToAPI = async (fishData) => {
    try {
      const response = await fetch('https://my-fly-box-api.herokuapp.com/api/v1/catches', {
        method: 'POST',
        headers: {
          'Content-Type': 'application/json;charset=utf-8'
        },
        body: JSON.stringify(fishData),
      });
      const fishDataResponse = await response.json();
      return fishDataResponse;
    } catch (error) {
      console.error(error);
    }
  };
