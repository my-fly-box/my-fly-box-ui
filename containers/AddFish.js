import React, { Component } from "react";
import { StyleSheet, Text, View, TouchableOpacity } from 'react-native';
import { connect } from "react-redux";
import FishForm from '../components/FishForm'
import { clearFishEntry, addFish } from '../actions';
import { addFishToAPI } from "../ApiCalls";

class AddFish extends Component {
  constructor(props) {
    super(props);
  }

  addFish = () => {
    const currentFishEntry = this.props.currentFishEntry
    addFishToAPI(currentFishEntry)
      .then(data => {
        this.props.addFish(data.data);
        this.props.clearFishEntry();
      }
    )
  }

  render() {
    return (
      <View style={styles.container}>
        {/* working on getting camera integrated */}
        <TouchableOpacity style={styles.button}
            onPress = {() => {
              this.props.navigation.navigate('PhotoSelector');
              }}>
            <Text style={styles.button}>Upload Image</Text>
          </TouchableOpacity>

        <FishForm />
        
        <View style={styles.buttonContainer}>
          <TouchableOpacity style={styles.button}
            onPress = {() => {
              this.props.navigation.navigate('FishCaught');
              this.props.clearFishEntry();
              }}>
            <Text>Cancel</Text>
          </TouchableOpacity>
        
          <TouchableOpacity style={styles.button}
            onPress = {() => {
              this.addFish();
              this.props.navigation.navigate('FishCaught');
              this.props.clearFishEntry();
              }}>
            <Text>Submit</Text>
          </TouchableOpacity>
        </View>
      </View>
    )
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    paddingTop: 45,
    width: "100%",
    height:"100%",
    justifyContent: 'center',
  },
  buttonContainer: {
    width: "80%",
    height:"100%", 
    flex: 1, 
    flexDirection: 'row',
    alignSelf: "center",
  },
  button: {
    backgroundColor: 'white',
    color: 'white',
    flex: 1,
    alignSelf: "center",
    margin: 5,
    marginTop: "20%",
    height: "20%",
    width: "40%",
    justifyContent: 'center',
    alignItems: 'center',
    borderWidth: 2,
    borderColor: '#2A9D8F',
    borderRadius: 3,
    shadowColor: '#000',
    shadowOffset: { width: 0, height: 2 },
    shadowOpacity: 0.8,
    shadowRadius: 2,
  },
})

const mapStateToProps = state => ({
  currentFishEntry: state.currentFishEntry,
})

const mapDispatchToProps = (dispatch) => ({
  clearFishEntry: () => dispatch( clearFishEntry() ),
  addFish: data => dispatch( addFish(data) ),
})

export default connect(mapStateToProps, mapDispatchToProps)(AddFish);