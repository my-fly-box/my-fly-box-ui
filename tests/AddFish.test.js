import React from 'react';
import 'react-native'
import AddFish from '../containers/AddFish'
import { Provider } from 'react-redux'
import rootReducer from '../reducers/index'
import { createStore } from 'redux'
import renderer from 'react-test-renderer'
import { fetchFlies } from '../ApiCalls'

jest.mock('../ApiCalls')


describe('AddFish', () => {
  let mockFlyData
  beforeEach(() => {
    mockFlyData = [
      {
        "id": "18",
        "type": "fly",
        "attributes": {
            "id": 18,
            "name": "Copper John",
            "size": 16,
            "color": "Copper",
            "category": "Nymph",
            "amount": 3
        }
    }
    ]
  })

  it('renders the AddFish', async () => {
    const testStore = createStore(rootReducer)
    const testWrapper = <Provider store={testStore}><AddFish /></Provider>
    fetchFlies.mockResolvedValueOnce(mockFlyData)
    const snap = renderer.create(
      testWrapper
    ).toJSON();
    expect(snap).toMatchSnapshot();
  })
}) 