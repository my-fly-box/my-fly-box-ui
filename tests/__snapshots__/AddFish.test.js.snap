// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddFish renders the AddFish 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#f8f8ff",
      "flex": 1,
      "justifyContent": "center",
      "paddingTop": 45,
    }
  }
>
  <Text
    style={
      Object {
        "alignSelf": "center",
        "color": "#0b7d83",
        "fontFamily": "Helvetica Neue",
        "fontSize": 20,
        "fontWeight": "bold",
      }
    }
  >
    Add information about your fish:
  </Text>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "alignSelf": "center",
        "backgroundColor": "#f7841f",
        "borderColor": "#212326",
        "borderRadius": 5,
        "borderWidth": 0.2,
        "elevation": 3,
        "height": 50,
        "justifyContent": "center",
        "marginBottom": "7%",
        "marginHorizontal": "5%",
        "marginTop": "4%",
        "opacity": 1,
        "shadowColor": "#000",
        "shadowOffset": Object {
          "height": 1,
          "width": 0,
        },
        "shadowOpacity": 0.22,
        "shadowRadius": 2.22,
        "width": 130,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#ffffff",
          "fontFamily": "Helvetica-Bold",
          "fontSize": 15,
          "textAlign": "center",
        }
      }
    >
      Upload Image
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
        "marginVertical": "3%",
        "paddingTop": 10,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#212326",
          "fontSize": 14,
          "fontWeight": "500",
          "marginBottom": 5,
        }
      }
    >
      Species:
    </Text>
    <TextInput
      allowFontScaling={true}
      onChangeText={[Function]}
      placeholder="Enter Species Name"
      rejectResponderTermination={true}
      style={
        Object {
          "backgroundColor": "#ffffff",
          "borderColor": "#212326",
          "borderRadius": 5,
          "borderWidth": 0.2,
          "color": "#212326",
          "elevation": 3,
          "fontSize": 14,
          "height": 45,
          "marginBottom": 5,
          "shadowColor": "#000",
          "shadowOffset": Object {
            "height": 1,
            "width": 0,
          },
          "shadowOpacity": 0.22,
          "shadowRadius": 2.22,
          "textAlign": "center",
          "width": "70%",
        }
      }
      underlineColorAndroid="transparent"
      value=""
    />
    <Text
      style={
        Object {
          "color": "#212326",
          "fontSize": 14,
          "fontWeight": "500",
          "marginBottom": 5,
        }
      }
    >
      Length of the Fish:
    </Text>
    <TextInput
      allowFontScaling={true}
      keyboardType="number-pad"
      onChangeText={[Function]}
      placeholder="Enter Length (in)"
      rejectResponderTermination={true}
      style={
        Object {
          "backgroundColor": "#ffffff",
          "borderColor": "#212326",
          "borderRadius": 5,
          "borderWidth": 0.2,
          "color": "#212326",
          "elevation": 3,
          "fontSize": 14,
          "height": 45,
          "marginBottom": 5,
          "shadowColor": "#000",
          "shadowOffset": Object {
            "height": 1,
            "width": 0,
          },
          "shadowOpacity": 0.22,
          "shadowRadius": 2.22,
          "textAlign": "center",
          "width": "70%",
        }
      }
      underlineColorAndroid="transparent"
      value=""
    />
    <Text
      style={
        Object {
          "color": "#212326",
          "fontSize": 14,
          "fontWeight": "500",
          "marginBottom": 5,
        }
      }
    >
      Weight of the Fish:
    </Text>
    <TextInput
      allowFontScaling={true}
      keyboardType="number-pad"
      onChangeText={[Function]}
      placeholder="Enter Weight (lb)"
      rejectResponderTermination={true}
      style={
        Object {
          "backgroundColor": "#ffffff",
          "borderColor": "#212326",
          "borderRadius": 5,
          "borderWidth": 0.2,
          "color": "#212326",
          "elevation": 3,
          "fontSize": 14,
          "height": 45,
          "marginBottom": 5,
          "shadowColor": "#000",
          "shadowOffset": Object {
            "height": 1,
            "width": 0,
          },
          "shadowOpacity": 0.22,
          "shadowRadius": 2.22,
          "textAlign": "center",
          "width": "70%",
        }
      }
      underlineColorAndroid="transparent"
      value=""
    />
    <Text
      style={
        Object {
          "color": "#212326",
          "fontSize": 14,
          "fontWeight": "500",
          "marginBottom": 5,
        }
      }
    >
      Location Caught:
    </Text>
    <TextInput
      allowFontScaling={true}
      onChangeText={[Function]}
      placeholder="Enter Location"
      rejectResponderTermination={true}
      style={
        Object {
          "backgroundColor": "#ffffff",
          "borderColor": "#212326",
          "borderRadius": 5,
          "borderWidth": 0.2,
          "color": "#212326",
          "elevation": 3,
          "fontSize": 14,
          "height": 45,
          "marginBottom": 5,
          "shadowColor": "#000",
          "shadowOffset": Object {
            "height": 1,
            "width": 0,
          },
          "shadowOpacity": 0.22,
          "shadowRadius": 2.22,
          "textAlign": "center",
          "width": "70%",
        }
      }
      underlineColorAndroid="transparent"
      value=""
    />
    <Text
      style={
        Object {
          "color": "#212326",
          "fontSize": 14,
          "fontWeight": "500",
          "marginBottom": 5,
        }
      }
    >
      Fly Used:
    </Text>
    <View
      style={
        Array [
          Object {
            "alignSelf": "stretch",
          },
          undefined,
        ]
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
        testID="ios_touchable_wrapper"
      >
        <View
          pointerEvents="box-only"
        >
          <TextInput
            allowFontScaling={true}
            editable={false}
            rejectResponderTermination={true}
            style={
              Array [
                Object {
                  "alignSelf": "center",
                  "backgroundColor": "white",
                  "borderColor": "#212326",
                  "borderRadius": 5,
                  "borderWidth": 0.2,
                  "elevation": 3,
                  "fontSize": 14,
                  "height": 45,
                  "paddingHorizontal": 10,
                  "paddingVertical": 12,
                  "shadowOffset": Object {
                    "height": 1,
                    "width": 0,
                  },
                  "shadowOpacity": 0.22,
                  "shadowRadius": 2.22,
                  "textAlign": "center",
                  "width": "70%",
                },
                Object {
                  "color": "#c7c7cd",
                },
              ]
            }
            testID="text_input"
            underlineColorAndroid="transparent"
            value="Select the Fly Used"
          />
        </View>
      </View>
      <Modal
        hardwareAccelerated={false}
        onOrientationChange={[Function]}
        supportedOrientations={
          Array [
            "portrait",
            "landscape",
          ]
        }
        testID="ios_modal"
        transparent={true}
        visible={false}
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "flex": 1,
              "opacity": 1,
            }
          }
          testID="ios_modal_top"
        />
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "backgroundColor": "#f8f8f8",
                "borderTopColor": "#dedede",
                "borderTopWidth": 1,
                "flexDirection": "row",
                "height": 45,
                "justifyContent": "space-between",
                "paddingHorizontal": 10,
                "zIndex": 2,
              },
              undefined,
            ]
          }
          testID="input_accessory_view"
        >
          <View
            style={
              Array [
                Object {
                  "flexDirection": "row",
                },
                undefined,
              ]
            }
          >
            <View
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                }
              }
            >
              <View
                style={
                  Array [
                    Object {
                      "backgroundColor": "transparent",
                      "borderColor": "#a1a1a1",
                      "borderRightWidth": 1.5,
                      "borderTopWidth": 1.5,
                      "height": 15,
                      "width": 15,
                    },
                    undefined,
                    Object {
                      "marginLeft": 11,
                      "transform": Array [
                        Object {
                          "translateY": 4,
                        },
                        Object {
                          "rotate": "-45deg",
                        },
                      ],
                    },
                    undefined,
                    Object {},
                  ]
                }
              />
            </View>
            <View
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                }
              }
            >
              <View
                style={
                  Array [
                    Object {
                      "backgroundColor": "transparent",
                      "borderColor": "#a1a1a1",
                      "borderRightWidth": 1.5,
                      "borderTopWidth": 1.5,
                      "height": 15,
                      "width": 15,
                    },
                    undefined,
                    Object {
                      "marginLeft": 22,
                      "transform": Array [
                        Object {
                          "translateY": -5,
                        },
                        Object {
                          "rotate": "135deg",
                        },
                      ],
                    },
                    undefined,
                    Object {},
                  ]
                }
              />
            </View>
          </View>
          <View
            accessible={true}
            focusable={true}
            hitSlop={
              Object {
                "bottom": 4,
                "left": 4,
                "right": 4,
                "top": 4,
              }
            }
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
            testID="done_button"
          >
            <View
              testID="needed_for_touchable"
            >
              <Text
                allowFontScaling={false}
                style={
                  Array [
                    Object {
                      "color": "#007aff",
                      "fontSize": 17,
                      "fontWeight": "600",
                      "paddingRight": 11,
                      "paddingTop": 1,
                    },
                    undefined,
                    Object {},
                  ]
                }
                testID="done_text"
              >
                Done
              </Text>
            </View>
          </View>
        </View>
        <View
          style={
            Array [
              Object {
                "backgroundColor": "#d0d4da",
                "justifyContent": "center",
              },
              Object {
                "height": 215,
              },
              undefined,
            ]
          }
        >
          <View>
            <RCTPicker
              items={
                Array [
                  Object {
                    "label": "Select the Fly Used",
                    "textColor": undefined,
                    "value": "Select the Fly Used",
                  },
                ]
              }
              onChange={[Function]}
              selectedIndex={0}
              style={
                Array [
                  Object {
                    "height": 216,
                  },
                  undefined,
                ]
              }
              testID="ios_picker"
            />
          </View>
        </View>
      </Modal>
    </View>
  </View>
  <View
    style={
      Object {
        "alignSelf": "center",
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "center",
        "width": "80%",
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "center",
          "backgroundColor": "#f7841f",
          "borderColor": "#212326",
          "borderRadius": 5,
          "borderWidth": 0.2,
          "elevation": 3,
          "height": 50,
          "justifyContent": "center",
          "marginHorizontal": "5%",
          "opacity": 1,
          "shadowColor": "#000",
          "shadowOffset": Object {
            "height": 1,
            "width": 0,
          },
          "shadowOpacity": 0.22,
          "shadowRadius": 2.22,
          "width": 130,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#ffffff",
            "fontFamily": "Helvetica-Bold",
            "fontSize": 15,
            "textAlign": "center",
          }
        }
      >
        Cancel
      </Text>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "center",
          "backgroundColor": "#f7841f",
          "borderColor": "#212326",
          "borderRadius": 5,
          "borderWidth": 0.2,
          "elevation": 3,
          "height": 50,
          "justifyContent": "center",
          "marginHorizontal": "5%",
          "opacity": 1,
          "shadowColor": "#000",
          "shadowOffset": Object {
            "height": 1,
            "width": 0,
          },
          "shadowOpacity": 0.22,
          "shadowRadius": 2.22,
          "width": 130,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#ffffff",
            "fontFamily": "Helvetica-Bold",
            "fontSize": 15,
            "textAlign": "center",
          }
        }
      >
        Submit
      </Text>
    </View>
  </View>
</View>
`;
